base/abstractarray.jl

```diff
@@ -1771,9 +1771,13 @@ typed_vcat(::Type{T}, A::AbstractVecOrMat...) where {T} = _typed_vcat(T, A)
 
 reduce(::typeof(vcat), A::AbstractVector{<:AbstractVecOrMat}) =
     _typed_vcat(mapreduce(eltype, promote_type, A), A)
+reduce(::typeof(vcat), A::Tuple{Vararg{AbstractVecOrMat}}) =
+    _typed_vcat(mapreduce(eltype, promote_type, A), A)
 
 reduce(::typeof(hcat), A::AbstractVector{<:AbstractVecOrMat}) =
     _typed_hcat(mapreduce(eltype, promote_type, A), A)
+reduce(::typeof(hcat), A::Tuple{Vararg{AbstractVecOrMat}}) =
```

Comment:
This avoids both `reduce(hcat, ())` (empty) and `reduce(hcat, ([1, 2]))` (currently returns a Vector as hcat is not called):
```suggestion
reduce(::typeof(hcat), A::Tuple{AbstractVecOrMat, AbstractVecOrMat, Vararg{AbstractVecOrMat}}) =
```